homeassistant/components/edl21/sensor.py

```diff
@@ -435,4 +450,17 @@ def extra_state_attributes(self):
     @property
     def native_unit_of_measurement(self):
         """Return the unit of measurement."""
-        return self._telegram.get("unit")
+        unit = self._telegram.get("unit")
+
+        if unit is None:
+            return None
+
+        if not (mapped_unit := SENSOR_UNIT_MAPPING.get(unit)):
+            _LOGGER.error(
+                "Unexpected unit of measurement %s detected. Please report at "
+                'https://github.com/home-assistant/core/issues?q=is%%3Aissue+label%%3A"integration%%3A+edl21"+',
+                unit,
+            )
+            return None
```

Comment:
Why not add them to the entity descriptions?

Reply:
In the edl21 component, the supported unit of measurement is being sent by the smart meter as part of the telegrams. So, setting the unit of measurement in the list of SensorEntityDescription will have no effect.

Reply:
But we know the unit of measurement for a voltage sensor will always be `V`?

Reply:
If we know that a certain obis always has a specific unit of measurement we could set the unit in the entity description. Otherwise that's not possible I think.

Reply:
I don't know whether certain obis always return a specific unit of measurement.
Thus, I think we should "start" with this minimal unit mapping and extend it as needed.